From 6801658497bfdf3ddac15e39ad7832784f750309 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Fri, 21 Jun 2024 14:23:43 +0800
Subject: [PATCH 0857/1448] riscv: hwprobe: add Zicond, Zacas and Ztso support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

  sync from lastest version commit info is:
     Merge patch series "riscv: hwprobe: add Zicond, Zacas and Ztso support"

    Clément Léger <cleger@rivosinc.com> says:

    This series add support for a few more extensions that are present in
    the RVA22U64/RVA23U64 (either mandatory or optional) and that are useful
    for userspace:
    - Zicond
    - Zacas
    - Ztso

    Series currently based on riscv/for-next.

    * b4-shazam-lts:
      riscv: hwprobe: export Zicond extension
      riscv: hwprobe: export Zacas ISA extension
      riscv: add ISA extension parsing for Zacas
      dt-bindings: riscv: add Zacas ISA extension description
      riscv: hwprobe: export Ztso ISA extension
      riscv: add ISA extension parsing for Ztso

    Link: https://lore.kernel.org/r/20231220155723.684081-1-cleger@rivosinc.com
    Signed-off-by: Palmer Dabbelt <palmer@rivosinc.com>

Change-Id: I6969dad287006f5e7d038b1633866c5993f3d1a5
---
 arch/riscv/kernel/sys_hwprobe.c | 37 +++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/riscv/kernel/sys_hwprobe.c b/arch/riscv/kernel/sys_hwprobe.c
index c4bc5c3081b4..969ef3d59dbe 100644
--- a/arch/riscv/kernel/sys_hwprobe.c
+++ b/arch/riscv/kernel/sys_hwprobe.c
@@ -96,6 +96,43 @@ static void hwprobe_isa_ext0(struct riscv_hwprobe *pair,
 		EXT_KEY(ZBB);
 		EXT_KEY(ZBS);
 		EXT_KEY(ZICBOZ);
+		EXT_KEY(ZBC);
+
+		EXT_KEY(ZBKB);
+		EXT_KEY(ZBKC);
+		EXT_KEY(ZBKX);
+		EXT_KEY(ZKND);
+		EXT_KEY(ZKNE);
+		EXT_KEY(ZKNH);
+		EXT_KEY(ZKSED);
+		EXT_KEY(ZKSH);
+		EXT_KEY(ZKT);
+		EXT_KEY(ZIHINTNTL);
+		EXT_KEY(ZTSO);
+		EXT_KEY(ZACAS);
+		EXT_KEY(ZICOND);
+		EXT_KEY(ZIHINTPAUSE);
+
+		if (has_vector()) {
+			EXT_KEY(ZVBB);
+			EXT_KEY(ZVBC);
+			EXT_KEY(ZVKB);
+			EXT_KEY(ZVKG);
+			EXT_KEY(ZVKNED);
+			EXT_KEY(ZVKNHA);
+			EXT_KEY(ZVKNHB);
+			EXT_KEY(ZVKSED);
+			EXT_KEY(ZVKSH);
+			EXT_KEY(ZVKT);
+			EXT_KEY(ZVFH);
+			EXT_KEY(ZVFHMIN);
+		}
+
+		if (has_fpu()) {
+			EXT_KEY(ZFH);
+			EXT_KEY(ZFHMIN);
+			EXT_KEY(ZFA);
+		}
 #undef EXT_KEY
 	}
 
-- 
2.47.0

